autoload/mucomplete/auto.vim

```diff
@@ -5,13 +5,36 @@
 let s:save_cpo = &cpo
 set cpo&vim
 
+fun! mucomplete#auto#start_timer() abort
+  if exists('s:completion_timer')
+    call mucomplete#auto#stop_timer()
+  endif
+
+  let l:delay = max([20, get(g:, 'mucomplete#delayed_completion', 0)])
+  let s:completion_timer = timer_start(l:delay, {-> mucomplete#auto#auto_complete() })
+endf
+
+fun! mucomplete#auto#stop_timer() abort
+  if !exists('s:completion_timer')
+    return
+  endif
+
+  call timer_stop(s:completion_timer)
+  unlet s:completion_timer
```

Comment:
I think you might invoke `timer_stop()` unconditionally. If the timer does not exists, there is no error.

Reply:
Probably, but i tested this only in neovim. Not sure how vim behaves. I'll give it a try.

Reply:
`timer_start` returns an id, which starts incrementally from 0. We must store it into variable because we could accidentally stop some other timer that user have. So even using some default value for `s:completion_timer` is not really an option.